app/communication_drivers/i2c_offboard_isolated/external_devices.c

```diff
@@ -25,7 +25,7 @@
 
 void init_i2c_offboard_external_devices(void)
 {
-    switch (g_ipc_mtoc.ps_module[0].ps_status.bit.model)
+    switch (g_ipc_mtoc.ps_module[g_current_ps_id].ps_status.bit.model)

```

Comment:
`g_current_ps_id` should not be used here, because it hasn't been initialized with any value. It's used only to identify ps modules on messages between ARM and DSP core. Please, keep it with `0`.